tests/run/freethreading_compatible.srctree

```diff
@@ -0,0 +1,40 @@
+PYTHON setup.py build_ext --inplace
+PYTHON -c "import default; default.test()"
+PYTHON -c "import compatible; compatible.test()"
+PYTHON -c "import incompatible; incompatible.test()"
```

Comment:
My view is:

we probably want most of this test to run in regular non-free-threading Python. We do want to be able to build and import modules that use this directive.

What I'd suggest is to have the `test` functions just return in non-freethreading builds and not actually do the test.